src/passes/OptimizeInstructions.cpp

```diff
@@ -2365,6 +2384,10 @@ struct OptimizeInstructions
         Bits::getMaxBits(left, this) == 1) {
       return builder.makeUnary(WrapInt64, left);
     }
+    // i64(x) != 0  ==>  !!x   (reduces 3 bytes into 2)
```

Comment:
Does this only apply to i64?

Reply:
Oh, very good point! Yes, this should apply to i32 as well. Fixed.